### DISCUSSION TITLE
Capability to run Pre-request Script and Tests

### DISCUSSION THREAD

**Discussion by cheungm1** [postmanlabs/postman-flows#181] [Ideas]
Usually, my Collection of APIs has Pre-request Script to create dynamic variables and store them in the Collection variables for later use. By using Flow, I try to convert my existing Collections of APIs but found out that the Flow cannot execute the script inside the Pre-request Script of the API, so it is not possible to convert it. If the Flow can have the ability to run the Pre-request Script for each API, then it will be very helpful for conversion.

Similarly, my Collection also has already written tests under the Tests section, if the Flow can also have the ability to run the tests and output the Pass or Fail result in a block, then it will be very helpful in conversion, otherwise, there will be a lot to do to re-create the validation tests.

**Comment by Bullkn0x** [comment]
Upvote!

**Comment by saswatds** [comment]
Hey @cheungm1 Pre-request Scripts and tests scripts do work in the flows, the only things that does't work is updating the environment and variables from scripts, and I have tried explaining that in this post https://github.com/postmanlabs/postman-flows/discussions/142

If you have tests written in your request, you can use the `Test Summary` block to see the result of the tests in the `Test Summary` pane.

![image](https://user-images.githubusercontent.com/8796447/150622751-1f7c1272-54f7-43be-b13e-78743ba0628c.png)
